nat-lab/tests/test_events_link_state.py

```diff
@@ -0,0 +1,489 @@
+                    wireguard=long_persistent_keepalive_periods(),
+                ),
+            )
+        )
+    ],
+)
+async def test_event_link_state_peers_idle_all_time(
+    alpha_setup_params: SetupParameters, beta_setup_params: SetupParameters
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        env = await setup_mesh_nodes(
+            exit_stack, [alpha_setup_params, beta_setup_params]
+        )
+        alpha, beta = env.nodes
+        client_alpha, client_beta = env.clients
+
+        # Expect no link event while peers are idle
+        await asyncio.sleep(20)
+        alpha_counts = client_beta.get_link_state_events_count(alpha.public_key)
+        beta_counts = client_alpha.get_link_state_events_count(beta.public_key)
+
+        # 1 down when node is Connecting and 1 up when node is Connected
+        assert alpha_counts == LinkStateEventsCount(none=0, up=1, down=1)
+        assert beta_counts == LinkStateEventsCount(none=0, up=1, down=1)
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "alpha_setup_params",
+    [
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_1,
+                adapter_type=AdapterType.LinuxNativeWg,
+                connection_tracker_config=generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_CONE_CLIENT_1,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+                features=TelioFeatures(
+                    no_link_detection=NoLinkDetection(rtt_seconds=1),
+                    wireguard=long_persistent_keepalive_periods(),
+                ),
+            )
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "beta_setup_params",
+    [
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_2,
+                adapter_type=AdapterType.LinuxNativeWg,
+                connection_tracker_config=generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_CONE_CLIENT_2,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+                features=TelioFeatures(
+                    no_link_detection=NoLinkDetection(rtt_seconds=1),
+                    wireguard=long_persistent_keepalive_periods(),
+                ),
+            )
+        )
+    ],
+)
+async def test_event_link_state_peers_exchanging_data_for_a_long_time(
+    alpha_setup_params: SetupParameters, beta_setup_params: SetupParameters
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        env = await setup_mesh_nodes(
+            exit_stack, [alpha_setup_params, beta_setup_params]
+        )
+        alpha, beta = env.nodes
+        client_alpha, client_beta = env.clients
+        connection_alpha, connection_beta = [
+            conn.connection for conn in env.connections
+        ]
+
+        for _ in range(0, 5):
```

Comment:
Correct me if I am wrong but the test executes 5 pings every second? This isn't a very long time(the testname says it is). I would have them chatting for at least a minute so to be sure that it's at least longer than the passive-keepalive(10s)+rtt